examples_gbs/run_tutorial_sample.py

```diff
@@ -0,0 +1,140 @@
+# pylint: disable=invalid-name,no-member,wrong-import-position,wrong-import-order,ungrouped-imports
+"""
+Sampling Tutorial
+=================
+
+This tutorial provides a quick guide to generating samples from GBS. This functionality is
+provided in the :mod:`~.gbs.sample` module of the applications layer.
+
+Sampling from GBS
+-----------------
+
+A GBS device can be programmed to sample from any symmetric matrix :math:`A`. To sample,
+we must specify both the mean number of photons being generated in the device and the form of
+detection used at the output: threshold detection or photon-number resolving (PNR) detection.
+Threshold detectors are restricted to measuring whether or not one or more photons have arrived
+at the detector, whereas PNR detectors are able to count the number of photons.
+
+Let's take a look at both types of sample. We can generate samples from a random 5-dimensional
+matrix:
+"""
+
+from strawberryfields.gbs import sample
+import numpy as np
+
+modes = 5
+n_mean = 4
+samples = 5
+
+A = np.random.random((modes, modes))
+A = A + A.T
+
+s_thresh = sample.sample(A, n_mean, samples, threshold=True)
+s_pnr = sample.sample(A, n_mean, samples, threshold=False)
+
+print(s_thresh)
+print(s_pnr)
+
+##############################################################################
+# In each case, a sample is a sequence of integers of ``len(modes) = 5``. Threshold samples are
+# ``0``'s and ``1``'s, corresponding to whether or not photons were detected in a mode. A ``1``
+# here is conventionally called a "click". PNR samples are non-negative integers counting the number
+# of photons detected in each mode. For example, suppose a PNR sample is ``[2, 1, 1, 0, 0]``,
+# meaning that 2 photons were detected in mode 0, 1 photons were detected in modes 1 and 2,
+# and 0 photons were detected in modes 3 and 4. If threshold detectors were used instead,
+# the sample would be: ``[1, 1, 1, 0, 0]``.
+#
+# Sampling subgraphs
+# ------------------
+#
+# So when would threshold detection or PNR detection be preferred in GBS? Since threshold samples
+# can be post-processed from PNR samples, we might expect that PNR detection is always the
+# preferred choice. However, *simulating* PNR-based GBS is significantly slower, and it turns out

```

Comment:
Maybe I'd say, "However, in practice *simulating*..."

Reply:
^agree with Nico. It's not obvious at this point in the tutorial why threshold sampling is faster, so we can just state that this is the case in practice